src/lib/Service/PermissionAwareLocationResolver.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformAdminUi\Service;
+
+use eZ\Publish\API\Repository\LocationService;
+use eZ\Publish\API\Repository\Values\Content\ContentInfo;
+use eZ\Publish\API\Repository\Values\Content\Location;
+
+final class PermissionAwareLocationResolver implements LocationResolver
+{
+    /** @var \eZ\Publish\API\Repository\LocationService */
+    private $locationService;
+
+    public function __construct(LocationService $locationService)
+    {
+        $this->locationService = $locationService;
+    }
+
+    /**
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
+     * @throws \eZ\Publish\API\Repository\Exceptions\BadStateException
+     */
+    public function resolveLocation(ContentInfo $contentInfo): Location
+    {
+        try {
+            $location = $this->locationService->loadLocation($contentInfo->mainLocationId);
+        } catch (\Exception $e) {
```

Comment:
Question: Should we limit this to only `UnauthorizedException | NotFoundException`? I don't think this is a good idea to silence other important exceptions i.e. `DatabaseException`?

Reply:
@webhdx resolved in https://github.com/ezsystems/ezplatform-admin-ui/pull/1351/commits/56f1cccdf7b5f842fccbc93c364f13e6973d6fbb